src/entry.php

```diff
@@ -1,24 +1,23 @@
 <?php namespace xp;
 
 function entry(&$argv) {
-  $ext= substr($argv[0], -4, 4);
-  if ('.php' === $ext) {
-    if (false === ($uri= realpath($argv[0]))) {
-      throw new \Exception('Cannot load '.$argv[0].' - does not exist');
+  if (is_file($argv[0])) {
```

Comment:
This additional stat call here is OK, as we'll be opening the file in any case shortly after this, and it will be cached at that time.